// @dasha.ai/sdk/dist/rest-api/generated/profiler/models/cluster-algorithm-enum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Profiler API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClusterAlgorithmEnum = void 0;
/**
 *
 * @export
 * @enum {string}
 */
var ClusterAlgorithmEnum;
(function (ClusterAlgorithmEnum) {
    ClusterAlgorithmEnum["Kmeans"] = "KMEANS";
    ClusterAlgorithmEnum["KmeansWithNode"] = "KMEANS_WITH_NODE";
    ClusterAlgorithmEnum["KmeansWithEdge"] = "KMEANS_WITH_EDGE";
    ClusterAlgorithmEnum["Dbscan"] = "DBSCAN";
    ClusterAlgorithmEnum["DbscanWithNode"] = "DBSCAN_WITH_NODE";
    ClusterAlgorithmEnum["DbscanWithEdge"] = "DBSCAN_WITH_EDGE";
})(ClusterAlgorithmEnum = exports.ClusterAlgorithmEnum || (exports.ClusterAlgorithmEnum = {}));
