// @dasha.ai/sdk/dist/rest-api/application.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationApi = void 0;
const errors = __importStar(require("../errors"));
const log_1 = require("../log");
const base = __importStar(require("./base"));
const api = __importStar(require("./generated/core/api"));
class ApplicationApi extends base.Api {
    constructor(account, log) {
        super(api.CustomerApplicationApi, account, log);
    }
    list() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.api.apiV1ApplicationsGet(undefined, undefined, {
                validateStatus: (status) => status < 400,
            });
            return response.data;
        });
    }
    getById(id) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.api.apiV1ApplicationsIdGet(id, {
                validateStatus: (status) => status < 400 || status === 404,
            });
            if (response.status === 404) {
                log_1.logAndThrow(this.log, new errors.ApplicationNotFoundError(id));
            }
            return response.data;
        });
    }
    getPackageById(id) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.api.apiV1ApplicationsIdPackageGet(id, {
                validateStatus: (status) => status < 400 || status === 404,
                responseType: "arraybuffer",
            });
            if (response.status === 404) {
                log_1.logAndThrow(this.log, new errors.ApplicationNotFoundError(id));
            }
            return {
                content: Buffer.from(response.data, "binary"),
                contentType: response.headers["content-type"],
            };
        });
    }
}
exports.ApplicationApi = ApplicationApi;
