// @dasha.ai/sdk/dist/rest-api/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TtsApi = exports.SipOutboundApi = exports.SipEndpointApi = exports.MiscApi = exports.GroupApi = exports.DiagnosticsApi = exports.ApplicationApi = exports.ProfilerApi = exports.CustomerApi = exports.models = void 0;
exports.models = __importStar(require("./models"));
var customer_1 = require("./customer");
Object.defineProperty(exports, "CustomerApi", { enumerable: true, get: function () { return customer_1.CustomerApi; } });
var profiler_1 = require("./profiler");
Object.defineProperty(exports, "ProfilerApi", { enumerable: true, get: function () { return profiler_1.ProfilerApi; } });
var application_1 = require("./application");
Object.defineProperty(exports, "ApplicationApi", { enumerable: true, get: function () { return application_1.ApplicationApi; } });
var diagnostics_1 = require("./diagnostics");
Object.defineProperty(exports, "DiagnosticsApi", { enumerable: true, get: function () { return diagnostics_1.DiagnosticsApi; } });
var group_1 = require("./group");
Object.defineProperty(exports, "GroupApi", { enumerable: true, get: function () { return group_1.GroupApi; } });
var misc_1 = require("./misc");
Object.defineProperty(exports, "MiscApi", { enumerable: true, get: function () { return misc_1.MiscApi; } });
var sip_endpoint_1 = require("./sip-endpoint");
Object.defineProperty(exports, "SipEndpointApi", { enumerable: true, get: function () { return sip_endpoint_1.SipEndpointApi; } });
var sip_outbound_1 = require("./sip-outbound");
Object.defineProperty(exports, "SipOutboundApi", { enumerable: true, get: function () { return sip_outbound_1.SipOutboundApi; } });
var tts_1 = require("./tts");
Object.defineProperty(exports, "TtsApi", { enumerable: true, get: function () { return tts_1.TtsApi; } });
