// @dasha.ai/sdk/dist/rest-api/sip-endpoint.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SipEndpointApi = void 0;
const errors = __importStar(require("../errors"));
const log_1 = require("../log");
const base = __importStar(require("./base"));
const api = __importStar(require("./generated/core/api"));
class SipEndpointApi extends base.Api {
    constructor(account, log) {
        super(api.SIPEndpointApi, account, log);
    }
    deleteConfig(name, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const getResponse = yield this.api.apiV1SipEndpointByNameGet(name, {
                validateStatus: (status) => status < 400 || status === 404,
            });
            if (getResponse.status === 404) {
                if (!(options === null || options === void 0 ? void 0 : options.quiet))
                    log_1.logAndThrow(this.log, new errors.SipConfigNotFoundError(name));
                return;
            }
            yield this.api.apiV1SipEndpointIdDelete(getResponse.data.id, {
                validateStatus: (status) => status < 400,
            });
        });
    }
    createConfig(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.api.apiV1SipEndpointPost(request, {
                validateStatus: (status) => status < 400 || status === 409,
            });
            if (response.status === 409) {
                log_1.logAndThrow(this.log, new errors.SipConfigExistsError(request.name));
            }
            return response.data;
        });
    }
    getConfigByName(name) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.api.apiV1SipEndpointByNameGet(name, {
                validateStatus: (status) => status < 400 || status === 404,
            });
            if (response.status === 404) {
                log_1.logAndThrow(this.log, new errors.SipConfigNotFoundError(name));
            }
            return response.data;
        });
    }
    listConfigs() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.api.apiV1SipEndpointGet(undefined, // skip
            undefined, // take
            {
                validateStatus: (status) => status < 400,
            });
            return response.data;
        });
    }
    updateConfig(name, request) {
        return __awaiter(this, void 0, void 0, function* () {
            const getResponse = yield this.api.apiV1SipEndpointByNameGet(name, {
                validateStatus: (status) => status < 400 || status === 404,
            });
            if (getResponse.status === 404) {
                log_1.logAndThrow(this.log, new errors.SipConfigNotFoundError(name));
            }
            const patchResponse = yield this.api.apiV1SipEndpointIdPatch(getResponse.data.id, request, {
                validateStatus: (status) => status < 400,
            });
            return patchResponse.data;
        });
    }
}
exports.SipEndpointApi = SipEndpointApi;
