// @dasha.ai/sdk/dist/grpc-api/generated/call-gateway_pb.js
"use strict";
// source: call-gateway.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.RequestCase', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.AudioSegment', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.Encoding', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.ResponseCase', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.MessageCase', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.Empty', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.InviteMessage', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.Type', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage', null, global);
goog.exportSymbol('proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.Reason', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.Empty = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.Empty, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.Empty.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.Empty';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.InviteMessage, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.InviteMessage';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.oneofGroups_);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.oneofGroups_);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.oneofGroups_);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.dasha.api.grpcapi.grpccallgateway.AudioSegment, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.displayName = 'proto.dasha.api.grpcapi.grpccallgateway.AudioSegment';
}
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.Empty.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.Empty.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.Empty} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.Empty.toObject = function (includeInstance, msg) {
        var f, obj = {};
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.Empty}
 */
proto.dasha.api.grpcapi.grpccallgateway.Empty.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.Empty;
    return proto.dasha.api.grpcapi.grpccallgateway.Empty.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.Empty} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.Empty}
 */
proto.dasha.api.grpcapi.grpccallgateway.Empty.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.Empty.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.Empty.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.Empty} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.Empty.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.toObject = function (includeInstance, msg) {
        var f, obj = {
            additionalInfo: jspb.Message.getFieldWithDefault(msg, 1, "")
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage;
    return proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {string} */ (reader.readString());
                msg.setAdditionalInfo(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getAdditionalInfo();
    if (f.length > 0) {
        writer.writeString(1, f);
    }
};
/**
 * optional string additional_info = 1;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.prototype.getAdditionalInfo = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectMessage.prototype.setAdditionalInfo = function (value) {
    return jspb.Message.setProto3StringField(this, 1, value);
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.toObject = function (includeInstance, msg) {
        var f, obj = {
            audioClientId: jspb.Message.getFieldWithDefault(msg, 1, 0)
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage;
    return proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {number} */ (reader.readUint64());
                msg.setAudioClientId(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getAudioClientId();
    if (f !== 0) {
        writer.writeUint64(1, f);
    }
};
/**
 * optional uint64 audio_client_id = 1;
 * @return {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.prototype.getAudioClientId = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};
/**
 * @param {number} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.prototype.setAudioClientId = function (value) {
    return jspb.Message.setProto3IntField(this, 1, value);
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.InviteMessage} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.toObject = function (includeInstance, msg) {
        var f, obj = {
            roomId: jspb.Message.getFieldWithDefault(msg, 1, ""),
            additionalInfo: jspb.Message.getFieldWithDefault(msg, 2, ""),
            url: jspb.Message.getFieldWithDefault(msg, 3, "")
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.InviteMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.InviteMessage;
    return proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.InviteMessage} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.InviteMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {string} */ (reader.readString());
                msg.setRoomId(value);
                break;
            case 2:
                var value = /** @type {string} */ (reader.readString());
                msg.setAdditionalInfo(value);
                break;
            case 3:
                var value = /** @type {string} */ (reader.readString());
                msg.setUrl(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.InviteMessage} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getRoomId();
    if (f.length > 0) {
        writer.writeString(1, f);
    }
    f = message.getAdditionalInfo();
    if (f.length > 0) {
        writer.writeString(2, f);
    }
    f = message.getUrl();
    if (f.length > 0) {
        writer.writeString(3, f);
    }
};
/**
 * optional string room_id = 1;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.prototype.getRoomId = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.InviteMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.prototype.setRoomId = function (value) {
    return jspb.Message.setProto3StringField(this, 1, value);
};
/**
 * optional string additional_info = 2;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.prototype.getAdditionalInfo = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.InviteMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.prototype.setAdditionalInfo = function (value) {
    return jspb.Message.setProto3StringField(this, 2, value);
};
/**
 * optional string url = 3;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.prototype.getUrl = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.InviteMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.prototype.setUrl = function (value) {
    return jspb.Message.setProto3StringField(this, 3, value);
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.toObject = function (includeInstance, msg) {
        var f, obj = {
            type: jspb.Message.getFieldWithDefault(msg, 1, 0),
            audioClientId: jspb.Message.getFieldWithDefault(msg, 2, 0),
            roomId: jspb.Message.getFieldWithDefault(msg, 3, "")
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage;
    return proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.Type} */ (reader.readEnum());
                msg.setType(value);
                break;
            case 2:
                var value = /** @type {number} */ (reader.readUint64());
                msg.setAudioClientId(value);
                break;
            case 3:
                var value = /** @type {string} */ (reader.readString());
                msg.setRoomId(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getType();
    if (f !== 0.0) {
        writer.writeEnum(1, f);
    }
    f = message.getAudioClientId();
    if (f !== 0) {
        writer.writeUint64(2, f);
    }
    f = message.getRoomId();
    if (f.length > 0) {
        writer.writeString(3, f);
    }
};
/**
 * @enum {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.Type = {
    JOINED: 0,
    LEFT: 1
};
/**
 * optional Type type = 1;
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.Type}
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.prototype.getType = function () {
    return /** @type {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.Type} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};
/**
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.Type} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.prototype.setType = function (value) {
    return jspb.Message.setProto3EnumField(this, 1, value);
};
/**
 * optional uint64 audio_client_id = 2;
 * @return {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.prototype.getAudioClientId = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};
/**
 * @param {number} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.prototype.setAudioClientId = function (value) {
    return jspb.Message.setProto3IntField(this, 2, value);
};
/**
 * optional string room_id = 3;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.prototype.getRoomId = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.prototype.setRoomId = function (value) {
    return jspb.Message.setProto3StringField(this, 3, value);
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.toObject = function (includeInstance, msg) {
        var f, obj = {
            roomId: jspb.Message.getFieldWithDefault(msg, 1, ""),
            reason: jspb.Message.getFieldWithDefault(msg, 2, 0),
            details: jspb.Message.getFieldWithDefault(msg, 3, "")
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage;
    return proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {string} */ (reader.readString());
                msg.setRoomId(value);
                break;
            case 2:
                var value = /** @type {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.Reason} */ (reader.readEnum());
                msg.setReason(value);
                break;
            case 3:
                var value = /** @type {string} */ (reader.readString());
                msg.setDetails(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getRoomId();
    if (f.length > 0) {
        writer.writeString(1, f);
    }
    f = message.getReason();
    if (f !== 0.0) {
        writer.writeEnum(2, f);
    }
    f = message.getDetails();
    if (f.length > 0) {
        writer.writeString(3, f);
    }
};
/**
 * @enum {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.Reason = {
    DISCONNECTED: 0,
    NOT_EXIST: 1,
    NOT_ALLOWED: 2
};
/**
 * optional string room_id = 1;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.prototype.getRoomId = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.prototype.setRoomId = function (value) {
    return jspb.Message.setProto3StringField(this, 1, value);
};
/**
 * optional Reason reason = 2;
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.Reason}
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.prototype.getReason = function () {
    return /** @type {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.Reason} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};
/**
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.Reason} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.prototype.setReason = function (value) {
    return jspb.Message.setProto3EnumField(this, 2, value);
};
/**
 * optional string details = 3;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.prototype.getDetails = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.prototype.setDetails = function (value) {
    return jspb.Message.setProto3StringField(this, 3, value);
};
/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.oneofGroups_ = [[1, 2, 3, 4]];
/**
 * @enum {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.MessageCase = {
    MESSAGE_NOT_SET: 0,
    CONNECTED: 1,
    INVITE: 2,
    TERMINATE: 3,
    ROOM_EVENT: 4
};
/**
 * @return {proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.MessageCase}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.getMessageCase = function () {
    return /** @type {proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.MessageCase} */ (jspb.Message.computeOneofCase(this, proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.oneofGroups_[0]));
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.toObject = function (includeInstance, msg) {
        var f, obj = {
            connected: (f = msg.getConnected()) && proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.toObject(includeInstance, f),
            invite: (f = msg.getInvite()) && proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.toObject(includeInstance, f),
            terminate: (f = msg.getTerminate()) && proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.toObject(includeInstance, f),
            roomEvent: (f = msg.getRoomEvent()) && proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.toObject(includeInstance, f)
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse;
    return proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = new proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage;
                reader.readMessage(value, proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.deserializeBinaryFromReader);
                msg.setConnected(value);
                break;
            case 2:
                var value = new proto.dasha.api.grpcapi.grpccallgateway.InviteMessage;
                reader.readMessage(value, proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.deserializeBinaryFromReader);
                msg.setInvite(value);
                break;
            case 3:
                var value = new proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage;
                reader.readMessage(value, proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.deserializeBinaryFromReader);
                msg.setTerminate(value);
                break;
            case 4:
                var value = new proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage;
                reader.readMessage(value, proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.deserializeBinaryFromReader);
                msg.setRoomEvent(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getConnected();
    if (f != null) {
        writer.writeMessage(1, f, proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage.serializeBinaryToWriter);
    }
    f = message.getInvite();
    if (f != null) {
        writer.writeMessage(2, f, proto.dasha.api.grpcapi.grpccallgateway.InviteMessage.serializeBinaryToWriter);
    }
    f = message.getTerminate();
    if (f != null) {
        writer.writeMessage(3, f, proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage.serializeBinaryToWriter);
    }
    f = message.getRoomEvent();
    if (f != null) {
        writer.writeMessage(4, f, proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage.serializeBinaryToWriter);
    }
};
/**
 * optional ConnectedMessage connected = 1;
 * @return {?proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.getConnected = function () {
    return /** @type{?proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage} */ (jspb.Message.getWrapperField(this, proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage, 1));
};
/**
 * @param {?proto.dasha.api.grpcapi.grpccallgateway.ConnectedMessage|undefined} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} returns this
*/
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.setConnected = function (value) {
    return jspb.Message.setOneofWrapperField(this, 1, proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.oneofGroups_[0], value);
};
/**
 * Clears the message field making it undefined.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.clearConnected = function () {
    return this.setConnected(undefined);
};
/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.hasConnected = function () {
    return jspb.Message.getField(this, 1) != null;
};
/**
 * optional InviteMessage invite = 2;
 * @return {?proto.dasha.api.grpcapi.grpccallgateway.InviteMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.getInvite = function () {
    return /** @type{?proto.dasha.api.grpcapi.grpccallgateway.InviteMessage} */ (jspb.Message.getWrapperField(this, proto.dasha.api.grpcapi.grpccallgateway.InviteMessage, 2));
};
/**
 * @param {?proto.dasha.api.grpcapi.grpccallgateway.InviteMessage|undefined} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} returns this
*/
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.setInvite = function (value) {
    return jspb.Message.setOneofWrapperField(this, 2, proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.oneofGroups_[0], value);
};
/**
 * Clears the message field making it undefined.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.clearInvite = function () {
    return this.setInvite(undefined);
};
/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.hasInvite = function () {
    return jspb.Message.getField(this, 2) != null;
};
/**
 * optional TerminateMessage terminate = 3;
 * @return {?proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.getTerminate = function () {
    return /** @type{?proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage} */ (jspb.Message.getWrapperField(this, proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage, 3));
};
/**
 * @param {?proto.dasha.api.grpcapi.grpccallgateway.TerminateMessage|undefined} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} returns this
*/
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.setTerminate = function (value) {
    return jspb.Message.setOneofWrapperField(this, 3, proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.oneofGroups_[0], value);
};
/**
 * Clears the message field making it undefined.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.clearTerminate = function () {
    return this.setTerminate(undefined);
};
/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.hasTerminate = function () {
    return jspb.Message.getField(this, 3) != null;
};
/**
 * optional RoomEventMessage room_event = 4;
 * @return {?proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.getRoomEvent = function () {
    return /** @type{?proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage} */ (jspb.Message.getWrapperField(this, proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage, 4));
};
/**
 * @param {?proto.dasha.api.grpcapi.grpccallgateway.RoomEventMessage|undefined} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} returns this
*/
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.setRoomEvent = function (value) {
    return jspb.Message.setOneofWrapperField(this, 4, proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.oneofGroups_[0], value);
};
/**
 * Clears the message field making it undefined.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.clearRoomEvent = function () {
    return this.setRoomEvent(undefined);
};
/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.dasha.api.grpcapi.grpccallgateway.ConnectionResponse.prototype.hasRoomEvent = function () {
    return jspb.Message.getField(this, 4) != null;
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.toObject = function (includeInstance, msg) {
        var f, obj = {
            roomId: jspb.Message.getFieldWithDefault(msg, 1, ""),
            additionalInfo: jspb.Message.getFieldWithDefault(msg, 2, "")
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage;
    return proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {string} */ (reader.readString());
                msg.setRoomId(value);
                break;
            case 2:
                var value = /** @type {string} */ (reader.readString());
                msg.setAdditionalInfo(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getRoomId();
    if (f.length > 0) {
        writer.writeString(1, f);
    }
    f = message.getAdditionalInfo();
    if (f.length > 0) {
        writer.writeString(2, f);
    }
};
/**
 * optional string room_id = 1;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.prototype.getRoomId = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.prototype.setRoomId = function (value) {
    return jspb.Message.setProto3StringField(this, 1, value);
};
/**
 * optional string additional_info = 2;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.prototype.getAdditionalInfo = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.InviteRequestMessage.prototype.setAdditionalInfo = function (value) {
    return jspb.Message.setProto3StringField(this, 2, value);
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.toObject = function (includeInstance, msg) {
        var f, obj = {
            roomId: jspb.Message.getFieldWithDefault(msg, 1, ""),
            additionalInfo: jspb.Message.getFieldWithDefault(msg, 2, "")
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage;
    return proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage}
 */
proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {string} */ (reader.readString());
                msg.setRoomId(value);
                break;
            case 2:
                var value = /** @type {string} */ (reader.readString());
                msg.setAdditionalInfo(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getRoomId();
    if (f.length > 0) {
        writer.writeString(1, f);
    }
    f = message.getAdditionalInfo();
    if (f.length > 0) {
        writer.writeString(2, f);
    }
};
/**
 * optional string room_id = 1;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.prototype.getRoomId = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.prototype.setRoomId = function (value) {
    return jspb.Message.setProto3StringField(this, 1, value);
};
/**
 * optional string additional_info = 2;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.prototype.getAdditionalInfo = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.LeaveMessage.prototype.setAdditionalInfo = function (value) {
    return jspb.Message.setProto3StringField(this, 2, value);
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.toObject = function (includeInstance, msg) {
        var f, obj = {
            roomId: jspb.Message.getFieldWithDefault(msg, 1, ""),
            sampleRate: jspb.Message.getFieldWithDefault(msg, 2, 0),
            encoding: jspb.Message.getFieldWithDefault(msg, 3, 0)
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest;
    return proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {string} */ (reader.readString());
                msg.setRoomId(value);
                break;
            case 2:
                var value = /** @type {number} */ (reader.readUint32());
                msg.setSampleRate(value);
                break;
            case 3:
                var value = /** @type {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.Encoding} */ (reader.readEnum());
                msg.setEncoding(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getRoomId();
    if (f.length > 0) {
        writer.writeString(1, f);
    }
    f = message.getSampleRate();
    if (f !== 0) {
        writer.writeUint32(2, f);
    }
    f = message.getEncoding();
    if (f !== 0.0) {
        writer.writeEnum(3, f);
    }
};
/**
 * @enum {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.Encoding = {
    LINEAR16: 0
};
/**
 * optional string room_id = 1;
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.prototype.getRoomId = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};
/**
 * @param {string} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.prototype.setRoomId = function (value) {
    return jspb.Message.setProto3StringField(this, 1, value);
};
/**
 * optional uint32 sample_rate = 2;
 * @return {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.prototype.getSampleRate = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};
/**
 * @param {number} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.prototype.setSampleRate = function (value) {
    return jspb.Message.setProto3IntField(this, 2, value);
};
/**
 * optional Encoding encoding = 3;
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.Encoding}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.prototype.getEncoding = function () {
    return /** @type {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.Encoding} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};
/**
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.Encoding} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.prototype.setEncoding = function (value) {
    return jspb.Message.setProto3EnumField(this, 3, value);
};
/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.oneofGroups_ = [[1, 2]];
/**
 * @enum {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.RequestCase = {
    REQUEST_NOT_SET: 0,
    INFO: 1,
    AUDIO: 2
};
/**
 * @return {proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.RequestCase}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.getRequestCase = function () {
    return /** @type {proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.RequestCase} */ (jspb.Message.computeOneofCase(this, proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.oneofGroups_[0]));
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.toObject = function (includeInstance, msg) {
        var f, obj = {
            info: (f = msg.getInfo()) && proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.toObject(includeInstance, f),
            audio: (f = msg.getAudio()) && proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.toObject(includeInstance, f)
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest;
    return proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = new proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest;
                reader.readMessage(value, proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.deserializeBinaryFromReader);
                msg.setInfo(value);
                break;
            case 2:
                var value = new proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment;
                reader.readMessage(value, proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.deserializeBinaryFromReader);
                msg.setAudio(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getInfo();
    if (f != null) {
        writer.writeMessage(1, f, proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest.serializeBinaryToWriter);
    }
    f = message.getAudio();
    if (f != null) {
        writer.writeMessage(2, f, proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.serializeBinaryToWriter);
    }
};
/**
 * optional AudioStreamRequest info = 1;
 * @return {?proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.getInfo = function () {
    return /** @type{?proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest} */ (jspb.Message.getWrapperField(this, proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest, 1));
};
/**
 * @param {?proto.dasha.api.grpcapi.grpccallgateway.AudioStreamRequest|undefined} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest} returns this
*/
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.setInfo = function (value) {
    return jspb.Message.setOneofWrapperField(this, 1, proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.oneofGroups_[0], value);
};
/**
 * Clears the message field making it undefined.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.clearInfo = function () {
    return this.setInfo(undefined);
};
/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.hasInfo = function () {
    return jspb.Message.getField(this, 1) != null;
};
/**
 * optional RawAudioSegment audio = 2;
 * @return {?proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.getAudio = function () {
    return /** @type{?proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment} */ (jspb.Message.getWrapperField(this, proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment, 2));
};
/**
 * @param {?proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment|undefined} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest} returns this
*/
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.setAudio = function (value) {
    return jspb.Message.setOneofWrapperField(this, 2, proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.oneofGroups_[0], value);
};
/**
 * Clears the message field making it undefined.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.clearAudio = function () {
    return this.setAudio(undefined);
};
/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioBidirectionalStreamRequest.prototype.hasAudio = function () {
    return jspb.Message.getField(this, 2) != null;
};
/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.oneofGroups_ = [[1, 2]];
/**
 * @enum {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.ResponseCase = {
    RESPONSE_NOT_SET: 0,
    ID: 1,
    AUDIO: 2
};
/**
 * @return {proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.ResponseCase}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.getResponseCase = function () {
    return /** @type {proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.ResponseCase} */ (jspb.Message.computeOneofCase(this, proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.oneofGroups_[0]));
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.toObject = function (includeInstance, msg) {
        var f, obj = {
            id: (f = msg.getId()) && proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.toObject(includeInstance, f),
            audio: (f = msg.getAudio()) && proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.toObject(includeInstance, f)
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse;
    return proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = new proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId;
                reader.readMessage(value, proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.deserializeBinaryFromReader);
                msg.setId(value);
                break;
            case 2:
                var value = new proto.dasha.api.grpcapi.grpccallgateway.AudioSegment;
                reader.readMessage(value, proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.deserializeBinaryFromReader);
                msg.setAudio(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getId();
    if (f != null) {
        writer.writeMessage(1, f, proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.serializeBinaryToWriter);
    }
    f = message.getAudio();
    if (f != null) {
        writer.writeMessage(2, f, proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.serializeBinaryToWriter);
    }
};
/**
 * optional AudioStreamId id = 1;
 * @return {?proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.getId = function () {
    return /** @type{?proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId} */ (jspb.Message.getWrapperField(this, proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId, 1));
};
/**
 * @param {?proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId|undefined} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse} returns this
*/
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.setId = function (value) {
    return jspb.Message.setOneofWrapperField(this, 1, proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.oneofGroups_[0], value);
};
/**
 * Clears the message field making it undefined.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.clearId = function () {
    return this.setId(undefined);
};
/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.hasId = function () {
    return jspb.Message.getField(this, 1) != null;
};
/**
 * optional AudioSegment audio = 2;
 * @return {?proto.dasha.api.grpcapi.grpccallgateway.AudioSegment}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.getAudio = function () {
    return /** @type{?proto.dasha.api.grpcapi.grpccallgateway.AudioSegment} */ (jspb.Message.getWrapperField(this, proto.dasha.api.grpcapi.grpccallgateway.AudioSegment, 2));
};
/**
 * @param {?proto.dasha.api.grpcapi.grpccallgateway.AudioSegment|undefined} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse} returns this
*/
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.setAudio = function (value) {
    return jspb.Message.setOneofWrapperField(this, 2, proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.oneofGroups_[0], value);
};
/**
 * Clears the message field making it undefined.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.clearAudio = function () {
    return this.setAudio(undefined);
};
/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamResponse.prototype.hasAudio = function () {
    return jspb.Message.getField(this, 2) != null;
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.toObject = function (includeInstance, msg) {
        var f, obj = {
            streamId: jspb.Message.getFieldWithDefault(msg, 1, 0)
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId;
    return proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {number} */ (reader.readUint32());
                msg.setStreamId(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getStreamId();
    if (f !== 0) {
        writer.writeUint32(1, f);
    }
};
/**
 * optional uint32 stream_id = 1;
 * @return {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.prototype.getStreamId = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};
/**
 * @param {number} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioStreamId.prototype.setStreamId = function (value) {
    return jspb.Message.setProto3IntField(this, 1, value);
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.toObject = function (includeInstance, msg) {
        var f, obj = {
            data: msg.getData_asB64()
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment}
 */
proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment;
    return proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment}
 */
proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {!Uint8Array} */ (reader.readBytes());
                msg.setData(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getData_asU8();
    if (f.length > 0) {
        writer.writeBytes(1, f);
    }
};
/**
 * optional bytes data = 1;
 * @return {!(string|Uint8Array)}
 */
proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.prototype.getData = function () {
    return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};
/**
 * optional bytes data = 1;
 * This is a type-conversion wrapper around `getData()`
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.prototype.getData_asB64 = function () {
    return /** @type {string} */ (jspb.Message.bytesAsB64(this.getData()));
};
/**
 * optional bytes data = 1;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getData()`
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.prototype.getData_asU8 = function () {
    return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(this.getData()));
};
/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.RawAudioSegment.prototype.setData = function (value) {
    return jspb.Message.setProto3BytesField(this, 1, value);
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.prototype.toObject = function (opt_includeInstance) {
        return proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioSegment} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.toObject = function (includeInstance, msg) {
        var f, obj = {
            streamId: jspb.Message.getFieldWithDefault(msg, 1, 0),
            time: jspb.Message.getFieldWithDefault(msg, 2, 0),
            data: msg.getData_asB64()
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioSegment}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.dasha.api.grpcapi.grpccallgateway.AudioSegment;
    return proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioSegment} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioSegment}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {number} */ (reader.readUint32());
                msg.setStreamId(value);
                break;
            case 2:
                var value = /** @type {number} */ (reader.readUint32());
                msg.setTime(value);
                break;
            case 3:
                var value = /** @type {!Uint8Array} */ (reader.readBytes());
                msg.setData(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.dasha.api.grpcapi.grpccallgateway.AudioSegment} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getStreamId();
    if (f !== 0) {
        writer.writeUint32(1, f);
    }
    f = message.getTime();
    if (f !== 0) {
        writer.writeUint32(2, f);
    }
    f = message.getData_asU8();
    if (f.length > 0) {
        writer.writeBytes(3, f);
    }
};
/**
 * optional uint32 stream_id = 1;
 * @return {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.prototype.getStreamId = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};
/**
 * @param {number} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioSegment} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.prototype.setStreamId = function (value) {
    return jspb.Message.setProto3IntField(this, 1, value);
};
/**
 * optional uint32 time = 2;
 * @return {number}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.prototype.getTime = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};
/**
 * @param {number} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioSegment} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.prototype.setTime = function (value) {
    return jspb.Message.setProto3IntField(this, 2, value);
};
/**
 * optional bytes data = 3;
 * @return {!(string|Uint8Array)}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.prototype.getData = function () {
    return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};
/**
 * optional bytes data = 3;
 * This is a type-conversion wrapper around `getData()`
 * @return {string}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.prototype.getData_asB64 = function () {
    return /** @type {string} */ (jspb.Message.bytesAsB64(this.getData()));
};
/**
 * optional bytes data = 3;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getData()`
 * @return {!Uint8Array}
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.prototype.getData_asU8 = function () {
    return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(this.getData()));
};
/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.dasha.api.grpcapi.grpccallgateway.AudioSegment} returns this
 */
proto.dasha.api.grpcapi.grpccallgateway.AudioSegment.prototype.setData = function (value) {
    return jspb.Message.setProto3BytesField(this, 3, value);
};
goog.object.extend(exports, proto.dasha.api.grpcapi.grpccallgateway);
